ipfsApi/client.py

```diff
@@ -201,16 +202,21 @@ def add(self, files, recursive=False, **kwargs):
         return self._add.request(self._client, (), files,
                                  recursive=recursive, **kwargs)
 
-    def get(self, multihash, **kwargs):
+    def get(self, multihash, output_path='.', **kwargs):
         """Downloads a file, or directory of files from IPFS.
 
-        Files are placed in the current working directory.
+        Files are placed in the current working directory if
+        output_path is not specified. Otherwise, they are
+        moved to output_path.
 
         Keyword arguments:
         multihash -- unique checksum used to identify IPFS resources
+        output_path -- where to put the downloaded file
         kwargs -- additional named arguments
         """
-        return self._get.request(self._client, multihash, **kwargs)
+        self._get.request(self._client, multihash, **kwargs)
+        if output_path is not '.':
+            shutil.move(multihash, output_path)
```

Comment:
> The other approach that I can see is to push this logic down into the DownloadCommand or maybe further into the http module. This could ensure that the file is never written to the CWD unless the user didn't specify another path. …
> 
> Do you see any other alternatives?

That is exactly what I would have done, so no. Just pass the parameter down the stack. It tuns out the `http.py` modules [already has support for passing in a path](https://github.com/ipfs/py-ipfs-api/blob/master/ipfsApi/http.py#L153) (the `filepath` paramter). However it appears that the tar archive always contains a single file or directory with the IPFS hash as name that will then contain the actual contents, so you'll probably need to adapt the function somewhat to iterate over all members of the tar separately and extract them to the correct path.
Look at the implementation of `tarfile.TarFile.extractall` for this; specifically the part were it iterates over the members of the archive and modifies the file mode of directories: Something similar should be possible to do with the actual file names retrieved as part of the tar archive.

I hope this gets you started on tackling this. :smiley: 

> Also, do you think that the module's current assumptions about the ability to write to the CWD represent a systemic design flaw?

No, extracting to the CWD with the hash as file name is a sane default for the quick-'n'-dirty scripts that are likely to use our library. It just doesn't make a lot of sense in more "advanced" (as in: people you actually know what they do) scenarios.

> Thanks so much for these code reviews, by the way.

Your welcome! :smiley_cat: